gradio/blocks.py

```diff
@@ -1354,7 +1359,10 @@ def handle_streaming_outputs(
         for i, output_id in enumerate(self.dependencies[fn_index]["outputs"]):
             block = self.blocks[output_id]
             if isinstance(block, StreamableOutput) and block.streaming:
-                stream = block.stream_output(data[i])
+                if end:
+                    stream = None
+                else:
+                    stream = block.stream_output(data[i])
```

Comment:
Why is this needed, isn't the end result of the iterator going to be None anyways?

Reply:
you're right! will remove